src/descriptor/mod.rs

```diff
@@ -231,18 +231,27 @@ impl MiniscriptKey for DescriptorPublicKey {
     type Hash = hash160::Hash;
 
     fn to_pubkeyhash(&self) -> Self::Hash {
+        self.to_public_key().to_pubkeyhash()
+    }
+}
+
+impl ToPublicKey for DescriptorPublicKey {
+    fn to_public_key(&self) -> bitcoin::PublicKey {
         match *self {
-            DescriptorPublicKey::PubKey(ref pk) => pk.to_pubkeyhash(),
+            DescriptorPublicKey::PubKey(ref pk) => *pk,
             DescriptorPublicKey::XPub(ref xpub) => {
```

Comment:
Is this is the expected behaviour for wildcard xpubs?

Reply:
I think so. All we do on derivation is appending to the internal derivation path (the one that still has to be applied) iif it's an xpub which is also marked as wildcard. So when calculating the actual pub key for an xpub we need to apply this derivation path (nothing to do in case of normal pubs, they might be some static backup key, idk, they definitely don't need any derivation conceptually).